### PR TITLE
Add a property to enable or disable the `PointerWheelEvent` on the `Calendar`-control

### LINKED ISSUE #6579: Calendar should have property to disable selection on mouse wheel
**Is your feature request related to a problem? Please describe.**
At the moment the `Calendar` changes the selection when the user rotates the mouse wheel. If the control is placed inside a `ScrollViewer` this may be changed by accident.

**Describe the solution you'd like**
 I would like to disable the mouse wheel selection behavior and instead just change the `DisplayDate`. I would add a new `StyledProperty` called `InterceptMouseWheelSelection [true|false]` which can be changed easily.   

**Describe alternatives you've considered**
1. Handle the mouse wheel by my own and set `e.Handled=true`
2.  overriding the control by my own 

**Additional context**
If you find the idea good I will create a PR for this.

### PR DESCRIPTION
**Pull request by timunie**
## What does the pull request do?
First of all this PR changes the way how the pointer wheel event is handled. Instead of selecting a different date it should only switch the displayed date. This prevents from accidently chaning the `SelectedDate`. Moreover the user can opt out this behavior in order to handle the `PointerWheelChanged-Event` on his own.

## What is the current behavior?
The `SelectedDate` will change if you turn the pointer wheel

## What is the updated/expected behavior with this PR?
Instead of chaning the `SelectedDate` only the `DisplayDate` should change if the pointer wheel was turned. Moreover the user should be able to handle the pointer wheel on his own.

## How was the solution implemented (if it's not obvious)?
- Changed the void `OnPointerWheelChanged` 
- Added a new `StyledProperty` named `SwitchPageOnPointerWheel` 

## Checklist

- [ ] Added unit tests (if possible)? 
- [X] Added XML documentation to any related classes? ► For the new property
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation ► Has to be done after this PR is marked as OK.

## Breaking changes

- Pointer wheel will behave differntly after this PR is merged

## Obsoletions / Deprecations

None

## Fixed issues
Fixes #6579

### CODE REVIEW

**Comment by grokys** [comment]
> Instead of selecting a different date it should only switch the displayed date. This prevents from accidently chaning the SelectedDate.

Yes, this is an excellent change - scrolling definitely should not change the selected date!

> Moreover the user can opt out this behavior in order to handle the PointerWheelChanged-Event on his own.

However this I'm not so sure about - I can't really think of any precedent for adding a property to disable the scrollwheel. Feels like a niche feature and moreover one that could easily be implemented by intercepting the tunnelling scrollwheel event if needed (common with other controls that handle the scroll wheel).

It's true that adding the property isn't exactly high-cost, but my concerns are more about consistency with other controls. Do we also add this property to `ScrollViewer`, `NumericUpDown` etc?

**Comment by timunie** [comment]
Hi @grokys 

first of all thank you for the feedback. But I think this change is not sooo easy to decide. Let me share my additional thoughts with you. 

> scrolling definitely should not change the selected date!

Yes, that was the root cause for me. But hitting the `PageUp` and `PageDown` keys still changes the selected date instead of just the display date. What is your feeling about this? Is this intended or also not intended? If not intended, I think I should also change this in the PR. 

 > I can't really think of any precedent for adding a property to disable the scrollwheel

For this please imagine the following situation: You have a long page within a `ScrollViewer`. The user spins the mouse wheel to scroll down. Most controls will just let the user scroll down, but as soon as a `Calendar` is under the pointer the scroll stops and instead the calender changes the displayed month. This is different for different controls. Maybe I should open a seperate issue for this and provide a sample App there. The solution may be an attached property instead which can then be used by other controls as well. 

To sum it up, I'll delete this property here and consider a different solution if needed. 

/cc @maxkatz6 

Happy coding
Tim

**Comment by grokys** [comment]
> The solution may be an attached property instead which can then be used by other controls as well.

That may indeed be a good solution!

> To sum it up, I'll delete this property here and consider a different solution if needed.

Yeah if you make this PR be just about not changing calendar selection with the scrollwheel that would be better.

**Comment by timunie** [comment]
I made a new PR for this as I think this solution was not the best.
